File: src/main/java/de/huddeldaddel/euler/Problem14.java
package de.huddeldaddel.euler;

import de.huddeldaddel.euler.math.CollatzSequence;
import de.huddeldaddel.euler.math.CollatzSequence;

public class Problem14 {

    public static void main(String[] args) {
        System.out.println(new Problem14().getResult());
    }

    public long getResult() {
        int startNumber = 0;
        long maxSequenceLength = 0;
        for(int i=1; i<1_000_000; i++) {
            final long sequenceLength = CollatzSequence.Companion.getLength(i);
            if(sequenceLength > maxSequenceLength) {
                maxSequenceLength = sequenceLength;
                startNumber = i;
            }
        }

        return startNumber;
    }

}

File: src/main/kotlin/de/huddeldaddel/euler/Problem14.kt
package de.huddeldaddel.euler

import de.huddeldaddel.euler.math.CollatzSequence

/**
 * Solution for https://projecteuler.net/problem=14
 */
fun main() {
    println(Problem14().result)
}

class Problem14 {

    val result: Long
        get() {
            var startNumber = 0
            var maxSequenceLength: Long = 0
            for (i in 1..999999) {
                val sequenceLength = CollatzSequence.getLength(i.toLong()).toLong()
                if (sequenceLength > maxSequenceLength) {
                    maxSequenceLength = sequenceLength
                    startNumber = i
                }
            }

            return startNumber.toLong()
        }

}
